src/Collection.php

```diff
@@ -398,6 +398,21 @@ public function sum($callback = null)
     }
 
     /**
+     * Chunk the underlying collection array.
+     *
+     * @param  int   $size
+     * @return static
+     */
+    public function chunk($size, $preserve_keys = false)
+    {
+        $chunks = [];
+        foreach (array_chunk($this->items, $size, $preserve_keys) as $chunk) {
+            $chunks[] = new static($chunk);
```

Comment:
Not sure if this makes sense. It's possible it could turn a normal array into a collection without that being known. Also not sure what would happen if this was an array of objects?